linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -647,8 +648,10 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
         }
 
       case Closure(arrow, captureParams, params, restParam, body, captureValues) =>
-        transformClosureCommon(arrow, captureParams, params, restParam, body,
-            captureValues.map(transformExpr))
+        trampoline {
+          transformClosureCommon(arrow, captureParams, params, restParam, body,
+              captureValues.map(transformExpr(_).toPreTransform))(finishTransform(isStat))
+        }
```

Comment:
If you get a `PreTransform` through `transformExpr(x).toPreTransform`, you are necessarily forcing everything in `x`. The shape you want here is to use `pretransformExprs` to get the `tcaptureValues`:
```suggestion
        trampoline {
          pretransformExprs(captureValues) { tcaptureValues =>
            transformClosureCommon(arrow, captureParams, params, restParam, body,
                tcaptureValues)(finishTransform(isStat))
          }
        }
```

Reply:
Aaaaahhhhh ok that makes sense, thank you. And with this change, it's fixed the `ApplyStatically` issue!! :tada: